tiledb/sm/query/query.cc

```diff
@@ -104,7 +105,15 @@ Query::Query(StorageManager* storage_manager, Array* array, URI fragment_uri)
     config_ = storage_manager->config();
 }
 
-Query::~Query() = default;
+Query::~Query() {
+  bool found = false;
+  bool use_malloc_trim = false;
+  const Status& st =
+      config_.get<bool>("sm.use_refactored_readers", &use_malloc_trim, &found);
```

Comment:
The results of this configuration call won't change between Query object lifetime.